db/migrate/20240601155348_dedup_item_requests_in_requests.rb

```diff
@@ -0,0 +1,47 @@
+class DedupItemRequestsInRequests < ActiveRecord::Migration[7.1]
+  disable_ddl_transaction!
+
+  def up
+    # There's some really gross request data in Jan 2023 and earlier
+    # that we don't want to get swept up in this migration. We'll need
+    # to fix it some day but for now we'll make recent-ish, well-formed
+    # data obey the dedup requirement, which will reduce weird edge
+    # cases when fulfilling requests + editing distributions.
+    start_date = Date.new(2023, 6, 1)
+
+    Request.where(created_at: start_date..).find_each do |request|
+      grouped_item_requests = request.item_requests.to_a.group_by(&:item_id)
```

Comment:
Probably want to do a `next` here if `grouped_item_requests.values.none? {|v| v.size > 1}` , which means we don't have to do any updating.